### PR TITLE
feat: SRS-361 Implement saveParcelDescriptionsForSite

### PR DESCRIPTION
**Pull request by bolny**
# Description

I find it much easier to review this PR with whitespace changes disabled: https://github.com/bcgov/nr-site-registry/pull/147/files?diff=unified&w=1

- Implements the `saveParcelDescriptionsForSite` method in the `parcelDescriptionService`. It splits the incoming payload into parcel descriptions to be added, updated, and deleted since the logic is different for each. The full implementation, including specs, is well over 2000 lines of code, so I will create a separate PR for the add, update, and delete methods so that they can be examined in isolation.
- I have updated the existing spec based on my learnings regarding testing async code in typescript. There shouldn't be any new failing tests.

Fixes # SRS-361 (Partially)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

For this PR I did not perform any manual testing. I have a set of graphql queries for the full implementation that I will share for the PRs that implement the actual adding, updating, and deleting logic.

- [x] Full unit test coverage for the new method

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-147-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-147-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-147-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-147-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts [hunk 1]
```diff
@@ -181,6 +182,69 @@ export class ParcelDescriptionsService {
     parcelDescriptions: ParcelDescriptionInputDTO[],
     userInfo: any,
   ) {
-    // TODO: Implementation in subsequent PR.
+    this.sitesLogger.log(
```
Comment (nikhila-aot) [inline backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts:185]: We had implemented standardize error handling, details can be found in PR 132. Please include it in the service layer methods. Example reference can be found in https://github.com/bcgov/nr-site-registry/blob/dev/backend/src/app/services/associatedSite/associatedSite.service.ts

**Comment by acoard-aot** [comment]
Closing PR, so we can re-open #143  as we're limited by storage in OpenShift. Feel free to re-open this in 20minutes if I forgot.

**Comment by acoard-aot** [comment]
Re-opening after #143 has successfully deployed. Note: The action for Deploy WILL FAIL currently, that is to be expected, as #143 is consuming all the storage quota.  We will hopefully have this resolved soon once we have a new temporary environment provisioned for us.

File: backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts (hunk 1 above)
Comment (nikhila-aot) [reply backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts:185]: @bolny - I do not see the try catch and HttpException getting thrown. Are you yet to make the change ?

**Review by nikhila-aot — APPROVED** [review]

**Pull request merged**